src/screens/LoginScreen/index.js

```diff
@@ -1,25 +1,58 @@
-import React, { memo, useCallback } from 'react';
+import React, { memo, useCallback, useState } from 'react';
 import { useDispatch } from 'react-redux';
+import { useStatus, LOADING } from '@rootstrap/redux-tools';
 import { Text, View, TouchableOpacity } from 'react-native';
 import { object } from 'prop-types';
+import { LoginManager, AccessToken } from 'react-native-fbsdk';
 
 import LoginForm from 'components/LoginForm';
-import { login } from 'actions/userActions';
+import ErrorView from 'components/common/ErrorView';
+import { login, facebookLogin } from 'actions/userActions';
 import strings from 'locale';
 import { LOGIN_SCREEN, SIGN_UP_SCREEN } from 'constants/screens';
+import { FACEBOOK_PERMISSIONS } from 'constants/common';
 import Ovals from 'components/common/Ovals';
 import styles from './styles';
 
 const LoginScreen = ({ navigation }) => {
+  const { error, status } = useStatus(facebookLogin);
+  const [facebookError, setFacebookError] = useState('');
   const dispatch = useDispatch();
   const loginRequest = useCallback(user => dispatch(login(user)), [dispatch]);
   const handleLogin = useCallback(() => navigation.push(SIGN_UP_SCREEN), [navigation]);
+  const performFacebookLogin = useCallback(accessToken => dispatch(facebookLogin(accessToken)), [
+    dispatch,
+  ]);
+  const handleFacebookLogin = () => {
```

Comment:
What do you think about creating a hook to handle facebook login, logout, and errors? We can check this together if you want 🙂

Reply:
Ok, great!

Reply:
Yeah, encapsulating logic in a hook is generally a good idea, it decluters the component and makes it that much more readable.